ui/panels-plugin/src/components/echarts-wrapper/EChartsWrapper.tsx

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useState, useLayoutEffect } from 'react';
+import { isFunction, isString } from 'lodash-es';
+import debounce from 'lodash/debounce';
+import { ECharts, EChartsCoreOption, init } from 'echarts/core';
+import { Box, SxProps, Theme } from '@mui/material';
+
+export type onEventFunction = () => void;
+
+export type onEventsType = Record<string, onEventFunction>;
+
+export interface EChartsWrapper {
+  sx: SxProps<Theme>;
+  option: EChartsCoreOption;
+  theme?: string;
+  onChartReady?: (instance: ECharts) => void;
+  onEvents?: onEventsType;
```

Comment:
Took part of this from [echarts-for-react](https://github.com/hustcc/echarts-for-react/), planning to improve these optional props when I refactor LineChart to use EChartsWrapper and implement chart themes